dccs/dccs_preprocessing_v1.R

```diff
@@ -0,0 +1,410 @@
+    testDatTrim <- testDatTrim[3:nrow(testDatTrim),]
+
+
+    #ANYSWITCH
+    #switch defined as above, comparison between switch group and nonswitch group
+
+    #subset the data for switch versus nonswitch trials, create new dataframes for each
+    testDatTrim_anySwitch <- testDatTrim[testDatTrim$switchStatus, ]
+    testDatTrim_anySwitchOPP <- testDatTrim[!testDatTrim$switchStatus, ]
+
+    #compute mean accuracy and standard error for switch versus nonswitch trials
+    anySwitch_meanACC <- mean(testDatTrim_anySwitch$key_resp_13.corr)
+    anySwitchOPP_meanACC <- mean(testDatTrim_anySwitchOPP$key_resp_13.corr)
+
+    anySwitch_seACC <- sd(testDatTrim_anySwitch$key_resp_13.corr)/sqrt(length(testDatTrim_anySwitch$key_resp_13.corr))
+    anySwitchOPP_seACC <- sd(testDatTrim_anySwitchOPP$key_resp_13.corr)/sqrt(length(testDatTrim_anySwitchOPP$key_resp_13.corr))
+
+    #subset the data for correct trials only, separately for switch versus nonswitch trials, creating new dataframes for each
+    testDatTrim_anySwitchCorr <- testDatTrim_anySwitch[testDatTrim_anySwitch$key_resp_13.corr==1, ]
+    testDatTrim_anySwitchOPPCorr <- testDatTrim_anySwitchOPP[testDatTrim_anySwitchOPP$key_resp_13.corr==1, ]
+
+    #for correct trials, for both switch and nonswitch groups, compute mean RT (raw and log-corrected), standard error, and median RT
+    anySwitchCorr_meanRT <- mean(testDatTrim_anySwitchCorr$key_resp_13.rt)
+    anySwitchOPPCorr_meanRT <- mean(testDatTrim_anySwitchOPPCorr$key_resp_13.rt)
+
+    anySwitchCorr_seRT <- sd(testDatTrim_anySwitchCorr$key_resp_13.rt)/sqrt(length(testDatTrim_anySwitchCorr$key_resp_13.rt))
+    anySwitchOPPCorr_seRT <- sd(testDatTrim_anySwitchOPPCorr$key_resp_13.rt)/sqrt(length(testDatTrim_anySwitchOPPCorr$key_resp_13.rt))
+
+    anySwitchCorr_logMeanRT <- mean(log((1+testDatTrim_anySwitchCorr$key_resp_13.rt)))
+    anySwitchOPPCorr_logMeanRT <- mean(log((1+testDatTrim_anySwitchOPPCorr$key_resp_13.rt)))
+
+    anySwitchCorr_medianRT <- median(testDatTrim_anySwitchCorr$key_resp_13.rt)
+    anySwitchOPPCorr_medianRT <- median(testDatTrim_anySwitchOPPCorr$key_resp_13.rt)
+
+    #compute switch cost scores for accuracy, RT, log-RT
+    anySwitch_costAcc <-  anySwitchOPP_meanACC - anySwitch_meanACC
+    anySwitch_costRT <- anySwitchCorr_meanRT - anySwitchOPPCorr_meanRT
+    anySwitch_costLogRT <- anySwitchCorr_logMeanRT - anySwitchOPPCorr_logMeanRT
+
+
+    #TOSHAPE
+    #switch defined as above, independent calculations for switch/nonswitch and switch/preswitch
+
+    #subset the data for switch, preswitch, and nonswitch trials
+    testDatTrim_toShapePairs <- subset(testDatTrim, (testDatTrim$cue=="SHAPE" | testDatTrim$nextCue=="SHAPE"))
+    if(testDatTrim_toShapePairs$cue[1]=="SHAPE"){
+      testDatTrim_toShapePairs <- testDatTrim_toShapePairs[2:length(testDatTrim_toShapePairs$cue),] #remove first row because no COLOR trial follows final SHAPE trial
+    }
```

Comment:
sorry I am unsure why no COLOR trial follows final SHAPE trial, could you please explain how this works?

Reply:
If the final trial of the set is SHAPE, then there will be no COLOR trial after it with which to compare.